SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -200,14 +204,119 @@ void USpatialNetDriver::OnConnected()
+	if (ServerConnection)
+	{
+		// If we know the GSM is already accepting players, simply spawn.
+		if (GlobalStateManager->bAcceptingPlayers)
+		{
+			PlayerSpawner->SendPlayerSpawnRequest();
+		}
+
+		// Bind a delegate to spawn this player if AcceptingPlayers changes to true.
+		auto PlayerSpawnerRef = TWeakObjectPtr<USpatialPlayerSpawner>(PlayerSpawner);
+		GlobalStateManager->AcceptingPlayersChanged.BindLambda([PlayerSpawnerRef](bool bAcceptingPlayers) {
+			if (bAcceptingPlayers)
+			{
+				// Send the player spawn commands with retries
+				PlayerSpawnerRef->SendPlayerSpawnRequest();
+			}
+		});
+
+		// Begin querying the state of the GSM so we know the state of AcceptingPlayers.
+		GlobalStateManager->QueryGSM(true /*bWithRetry*/);
+	}
+
+	// Here if we are a server and this is server travel (there is a snapshot to load) we want to load the snapshot.
+	if(!ServerConnection && !SnapshotToLoad.IsEmpty() && Cast<USpatialGameInstance>(GetWorld()->GetGameInstance())->bIsWorkerAuthorativeOverGSM)
+	{
+		UE_LOG(LogSpatialOSNetDriver, Warning, TEXT("Worker authoriative over the GSM is loading snapshot: %s"), *SnapshotToLoad);
+		SnapshotManager->LoadSnapshot(SnapshotToLoad);
+	}
+
+	if (!ServerConnection)
+	{
+		// If we already have authority over the GSM then toggle accepting players.
+		if(GlobalStateManager->bHasLiveMapAuthority)
+		{
+			GlobalStateManager->ToggleAcceptingPlayers(true);
+		}
+
+		// Set a delegate which toggles accepting players when we receive authority changes over the GSM.
+		GlobalStateManager->OnAuthorityChanged.BindUObject(GlobalStateManager, &UGlobalStateManager::ToggleAcceptingPlayers);
+	}
 }
 
 void USpatialNetDriver::OnConnectFailed(const FString& Reason)
 {
 	UE_LOG(LogSpatialOSNetDriver, Error, TEXT("Could not connect to SpatialOS. Reason: %s"), *Reason);
 }
 
+void USpatialNetDriver::SpatialProcessServerTravel(const FString& URL, bool bAbsolute, AGameModeBase* GameMode)
+{
+#if WITH_SERVER_CODE
+
+	GameMode->StartToLeaveMap();
+
+	// Force an old style load screen if the server has been up for a long time so that TimeSeconds doesn't overflow and break everything
+	bool bSeamless = (GameMode->bUseSeamlessTravel && GameMode->GetWorld()->TimeSeconds < 172800.0f); // 172800 seconds == 48 hours
+
+	FString NextMap;
+	if (URL.ToUpper().Contains(TEXT("?RESTART")))
+	{
+		NextMap = UWorld::RemovePIEPrefix(GameMode->GetOutermost()->GetName());
+	}
+	else
+	{
+		int32 OptionStart = URL.Find(TEXT("?"));
+		if (OptionStart == INDEX_NONE)
+		{
+			NextMap = URL;
+		}
+		else
+		{
+			NextMap = URL.Left(OptionStart);
+		}
+	}
+
+	FGuid NextMapGuid = UEngine::GetPackageGuid(FName(*NextMap), GameMode->GetWorld()->IsPlayInEditor());
+
+	// Notify clients we're switching level and give them time to receive.
+	FString URLMod = URL;
+	APlayerController* LocalPlayer = GameMode->ProcessClientTravel(URLMod, NextMapGuid, bSeamless, bAbsolute);
+
+	UE_LOG(LogGameMode, Warning, TEXT("SpatialServerTravel - Wiping the world"), *URL);
```

Comment:
`LogGameMode` is intentional?

Reply:
It is intentional but can be up for debate. Essentially `ProcessServerTravel` is a `GameModeBase` function so it makes sense for the log to be `LogGameMode`. I'm fine with updating it if you think it should match the class it's in.